File: Backend/src/main/kotlin/com/sorsix/backend/api/requests/PropertyRequest.kt
package com.sorsix.backend.api.requests

import com.sorsix.backend.api.dtos.PropertyImageDTO
import com.sorsix.backend.api.dtos.UserAccountDTO
import com.sorsix.backend.domain.entities.City
import com.sorsix.backend.domain.entities.PropertyType

data class PropertyRequest(
    val id: Long,
    var nightlyPrice: Double,
    var name: String,
    var guests: Int,
    var beds: Int,
    var bedrooms: Int,
    var bathrooms: Int,
    var isGuestFavorite: Boolean,
    var description: String,
    var address: String,
    var longitude: Double,
    var latitude: Double,
    var city: City,
    var propertyType: PropertyType,
    var images: List<PropertyImageDTO>,
)

File: Frontend/src/app/model/requests/PropertyRequest.ts
import {FormControl} from "@angular/forms";
import {PropertyType} from "../PropertyType";
import {City} from "../City";

export interface PropertyRequest {
  nightlyPrice: FormControl<Number>;
  name: FormControl<string>;
  guests: FormControl<Number>;
  beds: FormControl<Number>;
  bedrooms: FormControl<Number>;
  bathrooms: FormControl<Number>;
  isGuestFavorite: FormControl<Boolean>;
  description: FormControl<string>;
  address: FormControl<string>;
  longitude: FormControl<Number>;
  latitude: FormControl<Number>;
  city: FormControl<City>;
  propertyType: FormControl<PropertyType>;
  attributes: FormControl<Number[]>;
}
